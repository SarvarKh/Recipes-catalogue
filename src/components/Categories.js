const Categories = ({categories, handleClick}) => {
    return (
        <>
            <h1>Categories</h1>
            <div className="flex categories-container">
                {
                categories !== undefined ?
                    categories.map(category => (
                    <a onClick={() => handleClick(category.strCategory)} key={category.idCategory}>  
                        <div className="category">
                        <img src={category.strCategoryThumb} />
                        {category.strCategory}
                        <p className="category-desc">{category.strCategoryDescription}</p>
                        </div>
                    </a>
                    ))
                : <h2>Loading...</h2>
                
                }
            </div>
        </>
    )
}

export default Categories;