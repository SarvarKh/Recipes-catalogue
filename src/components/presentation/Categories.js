import PropTypes from 'prop-types';
import MapCategories from '../container/MapCategories';

const Categories = ({ handleClick }) => (
  <>
    <h1 className="titles">Categories</h1>
    <div className="flex categories-container">
      <MapCategories handleClick={handleClick} />
    </div>
  </>
);

export default Categories;

Categories.propTypes = {
  handleClick: PropTypes.func.isRequired,
};
