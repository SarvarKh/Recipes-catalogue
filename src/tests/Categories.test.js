import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Categories from '../components/presentation/Categories';

const categories = undefined;

const handleClick = () => (
    console.log('Test Categories and logging categories')
)

it('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><Categories categories={categories} handleClick={handleClick} /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});