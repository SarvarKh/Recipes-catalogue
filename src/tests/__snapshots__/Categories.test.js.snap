// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1
    className="titles"
  >
    Categories
  </h1>,
  <div
    className="flex categories-container"
  >
    <h2>
      Loading...
    </h2>
  </div>,
]
`;
